__tests__/app.test.js

```diff
@@ -241,4 +241,74 @@ describe("GET /api/reviews/:review_id/comments", () => {
             expect(response.body.msg).toBe("Bad request")
         })
 })
+})
+
+describe("POST /api/reviews/:review_id/comments", () => {
+    test("POST:201 sends the posted comment to the client", () => {
+        const testComment = {
+            username: "mallionaire",  body: "test comment"
+        }
+        return request(app)
+        .post("/api/reviews/1/comments")
+        .send(testComment)
+        .expect(201)
+        .then(({body: {comment}}) => {
+            console.log(comment)
+            expect(comment).toEqual(
+                expect.objectContaining({
+                comment_id: 7,
+                body: 'test comment',
+                review_id: 1,
+                author: 'mallionaire',
+                votes: 0,
+                created_at: expect.any(String)
+              }))
+        })
+    })
+    test("POST:404 sends an appropriate error message when given a valid but non-existent review id", () => {
+        const testComment = {
+            username: "mallionaire",  body: "test comment"
+        }
+        return request(app)
+        .post("/api/reviews/100000/comments")
+        .send(testComment)
+        .then((({body: {msg}}) => {
+            expect(msg).toBe("Review not found")
+        }))
+        })
+    // })
+    test("POST:400 sends an appropriate error message when given an invalid review id", () => {
+        const testComment = {
+            username: "testUsername",  body: "test comment"
+        }
+        return request(app)
+        .post("/api/reviews/banana/comments")
+        .send(testComment)
+        .then((({body: {msg}}) => {
+            expect(msg).toBe("Bad request")
+        }))
+    })
+    test("POST:400 sends an appropriate error message when given an empty request body", () => {
+        const testComment = {
+            username: "testUsername",  body: ""
+        }
+        return request(app)
+        .post("/api/reviews/1/comments")
+        .send(testComment)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe("Invalid comment")
+        })
+    })
+    test("POST:400 sends an appropriate error message when given a username not in users database", () => {
+        const testComment = {
+            username: "invalidUser",  body: "test comment"
+        }
+        return request(app)
+        .post("/api/reviews/1/comments")
+        .send(testComment)
+        .then((({body: {msg}}) => {
+            expect(msg).toBe("User not found")
+        }))
```

Comment:
to successfully post on this endpoint you need to have both the username and body properties present - can you test for what happens if one of these properties are missing completely?